### PR TITLE
support to increase x or y in version of x.y.z, not only z #539

### PR DESCRIPTION
**Pull request by ajarmoniuk**
Implementing the requested feature.

### CODE REVIEW

**Comment by slawekjaranowski** [comment]
I will move whole block from
```
if ( !removeSnapshot && nextSnapshot )
```

which is responsible to increment snapshot version to separate method - than we can test it by unit tests
It is only operations on Strings - ITs test is to heavy for it

**Comment by ajarmoniuk** [comment]
Suggestion implemented.

File: src/test/java/org/codehaus/mojo/versions/SetMojoTest.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            {
+                assertThat( getIncrementedVersion( "1.0.0", null ), is( "1.0.1-SNAPSHOT" ) );
+                assertThat( getIncrementedVersion( "1.0.0-SNAPSHOT", null ), is( "1.0.1-SNAPSHOT" ) );
+                assertThat( getIncrementedVersion( "1.0.0-SNAPSHOT", 1 ), is( "2.0.0-SNAPSHOT" ) );
+                assertThat( getIncrementedVersion( "1.0.0-SNAPSHOT", 2 ), is( "1.1.0-SNAPSHOT" ) );
+                assertThat( getIncrementedVersion( "1.0.0-SNAPSHOT", 3 ), is( "1.0.1-SNAPSHOT" ) );
+
+                try
+                {
+                    getIncrementedVersion( "1.0.0", 0 );
```
Comment (slawekjaranowski) [inline src/test/java/org/codehaus/mojo/versions/SetMojoTest.java:30]: should be separate test - another case for check

File: src/test/java/org/codehaus/mojo/versions/SetMojoTest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                    fail();
+                }
+                catch ( MojoExecutionException e )
+                {
+                    assertThat( e.getMessage(),
+                            containsString( "nextSnapshotIndexToIncrement cannot be less than 1" ) );
+                }
+                try
+                {
+                    getIncrementedVersion( "1.0.0", 4 );
```
Comment (slawekjaranowski) [inline src/test/java/org/codehaus/mojo/versions/SetMojoTest.java:40]: should be separate test - another case for check

**Comment by slawekjaranowski** [comment]
Next time  please - reformat code in separate PR ...  😄

File: src/test/java/org/codehaus/mojo/versions/SetMojoTest.java (hunk 2 above)
Comment (ajarmoniuk) [reply src/test/java/org/codehaus/mojo/versions/SetMojoTest.java:59]: done

**Review by slawekjaranowski — APPROVED** [review]

**Comment by slawekjaranowski** [comment]
Squashed in 540785721e8cb21be5e15dabf031b06eb1ee0125

**Pull request merged**